src/library/searchqueryparser.cpp

```diff
@@ -98,42 +144,23 @@ QString SearchQueryParser::getTextArgument(QString argument,
             argument = tokens->takeFirst();
         }
     }
-
-    // Deal with quoted arguments. If this token started with a quote, then
-    // search for the closing quote.
-    if (argument.startsWith("\"")) {
+    StringMatch mode = StringMatch::Contains;
+    bool equalRequested = false;
+    if (argument.startsWith("=")) {
+        // strip the '=' from the argument
         argument = argument.mid(1);
-
-        int quote_index = argument.indexOf("\"");
-        while (quote_index == -1 && tokens->length() > 0) {
-            argument += " " + tokens->takeFirst();
-            quote_index = argument.indexOf("\"");
-        }
-
-        if (quote_index == -1) {
-            // No ending quote found. Since we think they are going to close the
-            // quote eventually, treat the entire token list as the argument for
-            // now.
-            return argument;
-        }
-
-        // Stuff the rest of the argument after the quote back into tokens.
-        QString remaining = argument.mid(quote_index+1).trimmed();
-        if (remaining.size() != 0) {
-            tokens->push_front(remaining);
-        }
-
-        if (quote_index == 0) {
-            // We have found an explicit empty string ""
-            // return it as "" to distinguish it from an unfinished empty string
-            argument = kMissingFieldSearchTerm;
-        } else {
-            // Slice off the quote and everything after.
-            argument = argument.left(quote_index);
-        }
+        // TODO(ronso0) should 'tag:=string' really be equal to 'tag:="string"?
```

Comment:
I'd argue yes, because `string` is just a single word. so the quotes are unnecessary and thus these should be equal imo.

Reply:
True, makes sense. And it's consistent with `bpm:=...` which takes also a single value.